controllers/auth.js

```diff
@@ -0,0 +1,129 @@
+const userModelSchema = require("../model/user");
+const jwt = require("jsonwebtoken");
+const { Email } = require("../utils/Email");
+//signup user
+const signup = async (req, res) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+    const user = await userModelSchema.findOne({ email: email });
+    if (user) {
+      throw new Error("email already exist", 422);
+    }
+    const newUser = await userModelSchema.create({
+      firstName,
+      lastName,
+      email,
+      password,
+    });
+    return res.status(200).json({
+      registerUser: newUser,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      error: "user already exist!",
+    });
+  }
+};
+
+// forgot password page action
+const forgotPasswordAction = async (req, res, next) => {
+  try {
+    const { email } = req.body;
+    const user = await userModelSchema.findOne({ email: email });
```

Comment:
what happened if use does not exist with that email?

Reply:
@shubham-ch247 it should not be handled in catch statement. Handle that condition in try block itself